src/renderer/entities/staking/lib/validatorsService.ts

```diff
@@ -5,268 +5,308 @@ import { Data, Option } from '@polkadot/types';
-    const [stake, prefs] = await Promise.all([getValidatorsStake(api, era), getValidatorsPrefs(api, era)]);
-
-    return merge(stake, prefs);
-  };
-
-  /**
-   * Get validators with their identity, apy and slashing spans
-   */
-  const getValidatorsWithInfo = async (
-    chainId: ChainId,
-    api: ApiPromise,
-    era: EraIndex,
-    isLightClient?: boolean,
-  ): Promise<ValidatorMap> => {
-    const [stake, prefs] = await Promise.all([getValidatorsStake(api, era), getValidatorsPrefs(api, era)]);
-
-    const mergedValidators = merge(stake, prefs);
-
-    const [identity, apy, slashes] = await Promise.all([
-      getIdentities(api, Object.keys(mergedValidators), isLightClient),
-      getApy(api, Object.values(mergedValidators)),
-      getSlashingSpans(api, Object.keys(stake), era, isLightClient),
-    ]);
-
-    return merge(mergedValidators, apy, identity, slashes);
-  };
-
-  const getValidatorsStake = async (api: ApiPromise, era: EraIndex): Promise<ValidatorMap> => {
-    // HINT: uncomment if we need full lit of nominators (even those who doesn't get rewards)
-    // const data = await api.query.staking.erasStakers.entries(era);
-    const data = await api.query.staking.erasStakersClipped.entries(era);
-    const maxNominatorRewarded = getMaxNominatorRewarded(api);
-
-    return data.reduce((acc, [storageKey, type]) => {
-      const address = storageKey.args[1].toString();
-      const totalStake = type.total.toString();
-      const oversubscribed = type.others.length >= maxNominatorRewarded;
-      const nominators = type.others.map((n) => ({ who: n.who.toString(), value: n.value.toString() }));
-
-      const payload = { totalStake, address, oversubscribed, nominators, ownStake: type.own.toString() };
-
-      return { ...acc, [address]: payload };
-    }, {});
-  };
-
-  const getValidatorsPrefs = async (api: ApiPromise, era: EraIndex): Promise<ValidatorMap> => {
-    const data = await api.query.staking.erasValidatorPrefs.entries(era);
-
-    return data.reduce((acc, [storageKey, type]) => {
-      const address = storageKey.args[1].toString();
-      const commission = parseFloat(type.commission.toHuman() as string);
-      const payload = { address, commission, blocked: type.blocked.toHuman() };
+import { ValidatorMap } from './common/types';
 
-      return { ...acc, [address]: payload };
-    }, {});
-  };
-
-  const getDefaultValidatorsAmount = (api: ApiPromise): number => {
-    if (isKusamaChainId(api.genesisHash.toHex())) return KUSAMA_MAX_NOMINATORS;
+export const validatorsService = {
+  getValidatorsWithInfo,
+  getValidatorsList,
+  getMaxValidators,
+  getNominators,
+};
 
-    return DEFAULT_MAX_NOMINATORS;
-  };
+/**
+ * Get simple validators list
+ */
+async function getValidatorsList(api: ApiPromise, era: EraIndex): Promise<ValidatorMap> {
+  const validatorsStakeFn = isOldRuntimeForValidators(api) ? getValidatorsStake : getValidatorsStake_NEW;
```

Comment:
We can extract `isOldRuntimeForValidators(api) ? getValidatorsStake : getValidatorsStake_NEW` into a separate function to avoid dublication